### ISSUE TITLE
kubescape installed in first directory in $PATH under $HOME

### ISSUE THREAD

**Issue by craigbox** [kubescape/kubescape#1015]
```
craigbox@mac:/tmp$ echo $PATH
/Users/craigbox/bin/google-cloud-sdk/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/craigbox/.krew/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin
```
Install Kubescape:

```
craigbox@mac:/tmp$ curl -s https://raw.githubusercontent.com/kubescape/kubescape/master/install.sh | /bin/bash -x
+ set -e
+ getopts v: option
+ '[' -z ']'
+ RELEASE=latest/download
+ echo -e '\033[0;36mInstalling Kubescape...'
Installing Kubescape...
+ echo

+ BASE_DIR=/Users/craigbox/.kubescape
+ KUBESCAPE_EXEC=kubescape
+ KUBESCAPE_ZIP=kubescape.zip
++ uname -s
+ osName=Darwin
+ [[ Darwin == *\M\I\N\G\W* ]]
+ [[ Darwin == *\D\a\r\w\i\n* ]]
+ osName=macos
+ mkdir -p /Users/craigbox/.kubescape
+ OUTPUT=/Users/craigbox/.kubescape/kubescape
+ DOWNLOAD_URL=https://github.com/kubescape/kubescape/releases/latest/download/kubescape-macos-latest
+ curl --progress-bar -L https://github.com/kubescape/kubescape/releases/latest/download/kubescape-macos-latest -o /Users/craigbox/.kubescape/kubescape
######################################################################## 100.0%
+ SUDO=
++ id -u
+ '[' 501 -ne 0 ']'
++ which sudo
+ '[' -n /usr/bin/sudo ']'
+ SUDO=sudo
+ install_dir=/usr/local/bin
+ for pdir in '${PATH//:/ }'
+ edir=/Users/craigbox/bin/google-cloud-sdk/bin
+ [[ /Users/craigbox/bin/google-cloud-sdk/bin == /Users/craigbox/* ]]
+ install_dir=/Users/craigbox/bin/google-cloud-sdk/bin
+ mkdir -p /Users/craigbox/bin/google-cloud-sdk/bin
+ SUDO=
+ break
+ chmod +x /Users/craigbox/.kubescape/kubescape
+ rm -f /usr/local/bin/kubescape
+ cp /Users/craigbox/.kubescape/kubescape /Users/craigbox/bin/google-cloud-sdk/bin/kubescape
+ rm -rf /Users/craigbox/.kubescape/kubescape
+ echo

+ echo -e '\033[32mFinished Installation.'
Finished Installation.
+ echo -e '\033[0m'
```

The install script seems to split my $PATH, and then install into the first directory which is under $HOME?

```
craigbox@mac:/tmp$ which kubescape
/Users/craigbox/bin/google-cloud-sdk/bin/kubescape
```

**Comment by AvineshTripathi** [comment]
Hey I would like to work on this issue

**Comment by matthyx** [comment]
@AvineshTripathi please do !

**Comment by AvineshTripathi** [comment]
Hey @matthyx where do we want kubescape to install if not in first directory of $PATH?

**Comment by matthyx** [comment]
> Hey @matthyx where do we want kubescape to install if not in first directory of $PATH?

I think that's a question for @craigbox ?

**Comment by vladklokun** [comment]
> Hey @matthyx where do we want kubescape to install if not in first directory of $PATH?

The predominant installation path for Go binaries ([Go](https://go.dev/doc/install), [Helm](https://github.com/helm/helm/blob/76157c6d06f981a42bfbb9ec636bae5443cb2c8f/scripts/get-helm-3#L25), [Skaffold](https://skaffold.dev/docs/install/#standalone-binary)) seems to be `/usr/local/bin`. But that usually requires root privileges, so it would require us to change our installer a bit.

If we want the user to be able to `curl | bash` a script, we should really avoid requiring elevated privileges for that, so we could use `$HOME/.local/bin` or `$HOME/.local/bin/kubescape/` as a default installation path.

@dwertent What do you think?

**Comment by craigbox** [comment]
I think state of the art is "ask the user to put your directory in the $PATH".

https://krew.sigs.k8s.io/docs/user-guide/setup/install/
https://linkerd.io/2.12/getting-started/#step-1-install-the-cli

etc.

I would expect `$HOME/.kubescape` but if there's a dir in my path that's writeable, you could prompt the user to ask if they want it to go there.

(The default environment on Linux seems not to include anything under `$HOME`; I'm guessing bash on Mac is the same. Not sure about zsh.)

**Comment by Rahul-Kumar-prog** [comment]
Assign me please.

**Comment by AvineshTripathi** [comment]
@craigbox I am wondering why this logic was added https://github.com/kubescape/kubescape/blob/master/install.sh#L47-L55

**Comment by craigbox** [comment]
Hi team; just a reminder that being assigned to an issue doesn't actually mean anything.  What really matters is asking questions, writing design comments or documents, and actually submitting code.  Please don't ask to be assigned to an issue without having given us some indication on what you want to do to fix the issue.  Thanks!

**Comment by craigbox** [comment]
We can't set the $PATH in an install script unless we `source` the script and use `export`.
If we want Kubescape to be able to run after the script, it either needs to be put in the existing path, or the user has to do a manual step.  (Most are happy to do this.)

In most cases there is nowhere in the existing $PATH you can write to, probably by design.  

If you use sudo to put it in `/usr/local/bin` then you can't update it automatically.

**Comment by dipankardas011** [comment]
I may be wrong be we can use $HOME to install configs and store the binary to /usr/local/bin
If I am wrong do correct me😄

**Comment by Rahul-Kumar-prog** [comment]
did this problem get fixed ?

**Comment by thefool76** [comment]
I start working on it now.

**Comment by Turuk3100** [comment]
> We can't set the $PATH in an install script unless we `source` the script and use `export`.
> 
> If we want Kubescape to be able to run after the script, it either needs to be put in the existing path, or the user has to do a manual step.  (Most are happy to do this.)
> 
> 
> 
> In most cases there is nowhere in the existing $PATH you can write to, probably by design.  
> 
> 
> 
> If you use sudo to put it in `/usr/local/bin` then you can't update it automatically.

Oki€

**Comment by HollowMan6** [comment]
A fix is at #1147
